packages/web-api-scan-runner/src/tasks/notification-queue-message-sender.spec.ts

```diff
@@ -0,0 +1,201 @@
+        reports: null,
+        run: {
+            state: 'queued' as OnDemandPageScanRunState,
+            timestamp: 'timestamp',
+        },
+        priority: 1,
+        itemType: ItemType.onDemandPageScanRunResult,
+        id: 'id',
+        partitionKey: 'item-partitionKey',
+        batchRequestId: 'batch-id',
+    };
+
+    beforeEach(() => {
+        scanConfig = {
+            failedPageRescanIntervalInHours: 3,
+            maxScanRetryCount: 4,
+            maxSendNotificationRetryCount: 2,
+            minLastReferenceSeenInDays: 5,
+            pageRescanIntervalInDays: 6,
+            accessibilityRuleExclusionList: [],
+            scanTimeoutInMin: 1,
+        };
+
+        storageConfigStub = {
+            notificationQueue: 'test-notification-queue',
+        } as StorageConfig;
+
+        onDemandPageScanRunResultProviderMock = Mock.ofType(OnDemandPageScanRunResultProvider, MockBehavior.Strict);
+        queueMock = Mock.ofType(Queue);
+        loggerMock = Mock.ofType(MockableLogger);
+        serviceConfigMock = Mock.ofType(ServiceConfiguration);
+        serviceConfigMock
+            .setup(async s => s.getConfigValue('scanConfig'))
+            .returns(async () => scanConfig)
+            .verifiable(Times.once());
+        systemMock = Mock.ofInstance(System, MockBehavior.Strict);
+
+        processStub = {} as typeof process;
+        processStub.env = { batchJobId: 'job 1' };
+
+        dispatcher = new NotificationQueueMessageSender(
+            onDemandPageScanRunResultProviderMock.object,
+            serviceConfigMock.object,
+            storageConfigStub,
+            queueMock.object,
+            loggerMock.object,
+            processStub,
+            systemMock.object,
+        );
+    });
+
+    it('Enqueue Notification Succeeded', async () => {
+        const notification = generateNotification(notificationSenderMetadata.scanNotifyUrl, 'queued', null, 200);
+        setupUpdateScanRunResultCall(getRunningJobStateScanResult(notification));
+
+        systemMock
+            .setup(sm => sm.wait(5000))
+            .returns(async () => Promise.resolve())
+            .verifiable(Times.never());
+
+        queueMock
+            .setup(qm => qm.createMessage(storageConfigStub.notificationQueue, notificationSenderMetadata))
+            .returns(async () => Promise.resolve(true))
+            .verifiable(Times.once());
+
+        await dispatcher.sendNotificationMessage(notificationSenderMetadata);
+    });
+
+    it('Send Notification Failed', async () => {
+        const notification = generateNotification(
+            notificationSenderMetadata.scanNotifyUrl,
+            'queueFailed',
+            {
+                errorType: 'InternalError',
+                message: 'Failed to enqueue the notification!',
+            },
+            500,
+        );
+        setupUpdateScanRunResultCall(getRunningJobStateScanResult(notification));
+
+        systemMock
+            .setup(sm => sm.wait(1000))
```

Comment:
test at least with retry count 4, to test the wait intervals - 1000, 2000, 6000 millisec etc.